Examples/Swift/UserTrackingModesExample.swift

```diff
@@ -116,24 +118,37 @@ class UserLocationButton : UIButton {
     }
     
     func updateArrow(for mode: MGLUserTrackingMode) {
-        var stroke: CGColor
-        switch (mode) {
+        var arrowStrokeColor: CGColor
+        var arrowPosition: CGPoint
+        var arrowFillColor: CGColor
+        var arrowRotation: CGFloat
+        
+        switch mode {
         case .none:
-            stroke = UIColor.white.cgColor
+            arrowStrokeColor = UIColor.white.cgColor
```

Comment:
The “set it all, every time” approach here is perhaps easier to read than the old ternary, but it’s also unnecessarily verbose and duplicates logic in a way that will make it difficult to tweak.

Try refactoring the setting of these variables out into a function, with an eye to reducing the duplication of values.